lib/nile/src/Nile.ts

```diff
@@ -13,18 +14,62 @@ import {
   ConfigurationParameters,
 } from './generated/openapi/src/runtime';
 
+export class EventsApi {
+  static onCounter: number = 0;
+  entities: EntitiesApi;
+  timers: { [key: string]: NodeJS.Timer };
+
+  constructor(entities: EntitiesApi) {
+    this.entities = entities;
+    this.timers = {};
+  }
+
+  on(options: {type: string, seq: any}, listener: (event: InstanceEvent) => Promise<void>): number  {
+    const id = EventsApi.onCounter++;
+    var seq = options.seq;
+    const getEvents = async () => {
+      console.log('calling with seq:', seq)
+      const events = await this.entities.instanceEvents({
+        type: options.type,
+        workspace: this.entities.workspace,
```

Comment:
I believe the workspace should already be set on `nile` at this point, so you shouldn't need to override it. If we really need to have this, I would use an instance var to keep it and make it configurable.

Reply:
As I think about this, I think we should remove it entirely and not allow a developer to switch workspaces. Decide that the top level what workspace you want to be in.